src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3623,17 +3623,29 @@ private BoundExpression BindNullCoalescingAssignmentOperator(AssignmentExpressio
                 return GenerateNullCoalescingAssignmentBadBinaryOpsError(node, leftOperand, rightOperand, diagnostics);
             }
 
-            // If the type of the left is Nullable<A0>, and the right is the default literal, the default will be
-            // converted to the default of Nullable<A0>, null, and not the default of A0. This is likely unintended
-            // on the part of the user, so we warn
-            if (leftType.IsNullableType() && rightOperand.IsLiteralDefault())
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            if (leftType.IsNullableType())
             {
-                Error(diagnostics, ErrorCode.WRN_DefaultLiteralConvertedToNullIsNotIntended, node.Right, leftType.GetNullableUnderlyingType());
+                // If B is implicitly convertible to A0, then the result type of this expression is A0, except if B is dynamic.
```

Comment:
>except if B is dynamic. [](start = 104, length = 23)

Why the exception for dynamic? Seems irregular

Reply:
This is explicitly called out in the spec, it's the same behavior that `??` has.

---
In reply to: [293040246](https://github.com/dotnet/roslyn/pull/36329#discussion_r293040246) [](ancestors = 293040246)